examples/datawarehouse/__tests__/streamingInputTable.integration.test.ts

```diff
@@ -1,68 +1,76 @@
-import { spawn, ChildProcess } from 'child_process';
-jest.setTimeout(360000);
+import { resolve } from "path";
+import * as athena from "athena-client"
 
-beforeAll(done => {
-    const stackCreate = spawn('pulumi', ["stack", "init", "serverless-db.integration.test", '--cwd', './examples/datawarehouse']);
-    const onStackCreateComplete = (exitCode: number) => {
-        // TODO: make stack name unique (append some randomness)
-        // TODO: pass config via --config stringArray
-        // expect(exitCode).toEqual(0);
-        const stackSelect = spawn('pulumi', ["stack", "select", "serverless-db.integration.test", '--cwd', './examples/datawarehouse']);
-        monitorProcAndInitNext(stackSelect, onStackSelectComplete);
-    };
+import { PulumiRunner } from "../../../testing/integration";
 
-    monitorProcAndInitNext(stackCreate, onStackCreateComplete);
+jest.setTimeout(360000);
 
-    const onStackSelectComplete = (exitCode: number) => {
-        expect(exitCode).toBe(0);
-        const pulumiUp = spawn('pulumi', ['up', '--non-interactive', '--cwd', './examples/datawarehouse']);
-        monitorProcAndInitNext(pulumiUp, onUpComplete);
-    };
+let runner: PulumiRunner;
+const region = "us-west-2";
 
-    const onUpComplete = (exitCode: number) => {
-        expect(exitCode).toBe(0)
-        done()
+beforeAll(async () => {
+    const config: { [key: string]: string } = {
+        "aws:region": region,
+        "pulumi-serverless-db:shards": '{"prod": 8, "gamma": 4, "beta": 1}',
+        "pulumi-serverless-db:stage": "beta",
+        "pulumi-serverless-db:dev": "true"
     };
 
+    const pulumiProjDir = resolve("./examples/datawarehouse");
```

Comment:
Should this test have it's own project that only creates what is needed for the test?
Definitely something that can be done later, but just a thought.

Reply:
I agree with this. Right now this is sort of an integration test on the example. Likely we'll need to fork off at some point.